.github/workflows/test-lang-csharp.yml

```diff
@@ -63,6 +63,19 @@ jobs:
       - name: Test
         run: ./build.sh test
 
+      # Build and test against .NET 7
+      # .NET 7 is not released yet, however this is a good way to test if the project is ready for the release
+      # Once .NET 7 is officially released, this can be removed and 7.0.x can be used instead above
+      - name: Install .NET SDK 7.0 (pre-release)
+        uses: actions/setup-dotnet@v1
+        with:
+          include-prerelease: true
+          dotnet-version: |
```

Comment:
In c# we generate a nupkg which has all the supported  .NET SDK version compiled. Currently netcoreapp3.1, net5.0, net6.0, which are the MS officially supported versions. This is defined in `common.props` by the `DefaultExeTargetFrameworks` prop. So the matric approach does not work, because all the supported SDKs must be installed in the building/testing image and not just one specific version. If the nupkg would be only netstandard we could get away with one specific version, however avrogen tool must be compiled for all available versions. When that tool is installed, the matching framework SDK version is selected from the nupkg.
E.g. This is the reason I am eager for the 1.11.1 version, because until now, I cannot use avrogen in purely/only .NET 6 SDK environment.

So the build step is the same, however when the normal build is done, I install .NET SDK and build/test only for net7.0 to check if all is good.

Just fyi, once NEt 7 is released, `DefaultExeTargetFrameworks` will be `netcoreapp3.1;net6.0;net7.0` which will reflect the EOL status of the dotnet SDKs (https://dotnet.microsoft.com/en-us/download/dotnet)